src/biokbase/narrative/jobs/job.py

```diff
@@ -457,24 +446,25 @@ def output_state(self, state=None) -> dict:
             except Exception as e:
                 # Can't get viewer params
                 new_e = transform_job_exception(e)
-                state.update(
-                    {
-                        "status": "error",
-                        "errormsg": "Unable to build output viewer parameters!",
-                        "error": {
-                            "code": getattr(new_e, "code", -1),
-                            "source": getattr(new_e, "source", "JobManager"),
-                            "name": "App Error",
-                            "message": "Unable to build output viewer parameters",
-                            "error": "Unable to generate App output viewer!\nThe App appears to have completed successfully,\nbut we cannot construct its output viewer.\nPlease contact https://kbase.us/support for assistance.",
-                        },
-                    }
-                )
+                widget_info = {
+                    "status": "error",
+                    "errormsg": "Unable to build output viewer parameters!",
+                    "error": {
+                        "code": getattr(new_e, "code", -1),
+                        "source": getattr(new_e, "source", "JobManager"),
+                        "name": "App Error",
+                        "message": "Unable to build output viewer parameters",
+                        "error": "Unable to generate App output viewer!\nThe App appears to have completed successfully,\nbut we cannot construct its output viewer.\nPlease contact https://kbase.us/support for assistance.",
+                    },
+                }
+                # update timestamp if there was an error
+                state.update({"updated": int(time.time())})
 
         job_state = {
             "state": state,
             "widget_info": widget_info,
             "user": self.user,
+            "cell_id": self.cell_id,
```

Comment:
Add the `cell_id` to the top level output so that it will be easier if we switch the job comm channel to using cell IDs to address job-related messages.